presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -1010,6 +1011,51 @@ public void testTime(boolean insertWithPresto, ZoneId sessionZone)
         }
     }
 
+    @Test
+    public void testTime24()
+    {
+        try (TestTable testTable = new TestTable(
+                new JdbcSqlExecutor(postgreSqlServer.getJdbcUrl()),
+                "tpch.test_time_24",
+                "(a time(0), b time(3), c time(6))",
+                List.of(
+                        // "zero" row
+                        "TIME '00:00:00', TIME '00:00:00.000', TIME '00:00:00.000000'",
+                        // "max" row
+                        "TIME '23:59:59', TIME '23:59:59.999', TIME '23:59:59.999999'",
+                        // "24" row
+                        "TIME '24:00:00', TIME '24:00:00.000', TIME '24:00:00.000000'"))) {
+            // select
+            assertThat(query("SELECT a, b, c FROM " + testTable.getName()))
+                    .matches("VALUES " +
+                            "(TIME '00:00:00.000', TIME '00:00:00.000', TIME '00:00:00.000'), " +
+                            "(TIME '23:59:59.000', TIME '23:59:59.999', TIME '23:59:59.999'), " +
+                            "(TIME '23:59:59.999', TIME '23:59:59.999', TIME '23:59:59.999')");
```

Comment:
Is this the value returned by PostgresSQL's JDBC driver or are we somehow changing it on the connector side?

Reply:
This is what PostgreSQL JDBC returns upon `rs.getObject(column)` or `rs.getObject(column, LocalTime.class)` call.

For setup
```
test=# create table test_time_24(a time(0), b time(3), c time(6));
CREATE TABLE
test=# insert into test_time_24 values (TIME '00:00:00', TIME '00:00:00.000', TIME '00:00:00.000000');
INSERT 0 1
test=# insert into test_time_24 values (TIME '23:59:59', TIME '23:59:59.999', TIME '23:59:59.999999');
INSERT 0 1
test=# insert into test_time_24 values (TIME '24:00:00', TIME '24:00:00.000', TIME '24:00:00.000000');
INSERT 0 1
test=# select * from test_time_24;
    a     |      b       |        c
----------+--------------+-----------------
 00:00:00 | 00:00:00     | 00:00:00
 23:59:59 | 23:59:59.999 | 23:59:59.999999
 24:00:00 | 24:00:00     | 24:00:00
(3 rows)
```

```
-----------------
rs.getObject(1) = 00:00:00
rs.getString(1) = 00:00:00
rs.getObject(1, LocalTime.class) = 00:00
rs.getObject(2) = 00:00:00
rs.getString(2) = 00:00:00
rs.getObject(2, LocalTime.class) = 00:00
rs.getObject(3) = 00:00:00
rs.getString(3) = 00:00:00
rs.getObject(3, LocalTime.class) = 00:00
-----------------
rs.getObject(1) = 23:59:59
rs.getString(1) = 23:59:59
rs.getObject(1, LocalTime.class) = 23:59:59
rs.getObject(2) = 23:59:59
rs.getString(2) = 23:59:59.999
rs.getObject(2, LocalTime.class) = 23:59:59.999
rs.getObject(3) = 23:59:59
rs.getString(3) = 23:59:59.999999
rs.getObject(3, LocalTime.class) = 23:59:59.999999
-----------------
rs.getObject(1) = 00:00:00
rs.getString(1) = 24:00:00
rs.getObject(1, LocalTime.class) = 23:59:59.999999999
rs.getObject(2) = 00:00:00
rs.getString(2) = 24:00:00
rs.getObject(2, LocalTime.class) = 23:59:59.999999999
rs.getObject(3) = 00:00:00
rs.getString(3) = 24:00:00
rs.getObject(3, LocalTime.class) = 23:59:59.999999999
```

Thus, as you see, we can tell apart 24:00:00 case using `getString`.
However, my PR does not substantially change treatment of 24:00:00 values, so
if you want to change treatment of those values, I'd suggest a followup issue.

Reply:
No, I was just curious about the behavior. It seems very broken on PostgreSQL's part ;)